RetailCoder.VBE/Inspections/QuickFixes/PassParameterByReferenceQuickFix.cs

```diff
@@ -19,22 +19,19 @@ public PassParameterByReferenceQuickFix(ParserRuleContext context, QualifiedSele
 
         public override void Fix()
         {
-            var parameter = Context.GetText();
+            string pattern = "^\\s*" + Tokens.ByVal + "(\\s+)";
+            string rgxReplacement = Tokens.ByRef + "$1";
+            Regex rgx = new Regex(pattern);
 
-            var parts = parameter.Split(new char[]{' '},2);
-            if (1 != parts.GetUpperBound(0))
-            {
-                return;
-            }
-            parts[0] = parts[0].Replace(Tokens.ByVal, Tokens.ByRef);
-            var newContent = parts[0] + " " + parts[1];
+            var parameter = Context.GetText();
```

Comment:
The `Context` here should be an `ArgContext`:

    arg : (OPTIONAL whiteSpace)? ((BYVAL | BYREF) whiteSpace)? (PARAMARRAY whiteSpace)? unrestrictedIdentifier (whiteSpace? LPAREN whiteSpace? RPAREN)? (whiteSpace? asTypeClause)? (whiteSpace? argDefaultValue)?;

...which contains everything you need to know to rewrite the tokens.

The regex will *probably* work, but the *ideal*, safest solution would use a `TokenStreamRewriter` to swap the `BYVAL` token with a `BYREF` token, or just to insert a `BYREF` token where it belongs.